### ISSUE TITLE
VariableAnalysis.UnusedVariable - bug when using foreach

### ISSUE THREAD

**Issue by Chrico** [inpsyde/php-coding-standards#5]
I see following warnings with the latest 0.7.x - release:

```
Unused variable $filter.(VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable)
Unused variable $validator. (VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable)
Unused variable $stored_data. (VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable)
```

The code is following:

```php
/**
 * Add a single Element.
 *
 * @param ElementInterface     $element
 * @param FilterInterface[]    $filters
 * @param ValidatorInterface[] $validators
 */
public function addElement(ElementInterface $element, array $filters = [], array $validators = [])
{

	$this->form->add_element($element);

	foreach ($filters as $filter) {
		$this->form->add_filter($element->get_name(), $filter);
	}

	foreach ($validators as $validator) {
		$this->form->add_validator($element->get_name(), $validator);
	}
}
```

**Comment by gmazzap** [comment]
@Chrico It's a bug in VariableAnalysis which I reported already, see https://github.com/sirbrillig/phpcs-variable-analysis/issues/33

But at the moment (in master branch) *I removed completely VariableAnalysis* because too much false positives and phpStorm do a good job already to show wrong usages of variables. 

We can think about adding VariableAnalysis again later.

**Comment by gmazzap** [comment]
Because VariableAnalysis  is not in use anymore, we can close this for now.
